### PR TITLE
Add automated tag suggestions from event description

### LINKED ISSUE #72: Implement Automated Event Tagging via AI
When an organizer enters an event description, the app should suggest relevant tags (e.g., #coding, #music, #workshop) using a keyword extraction service. This helps in maintaining a clean and searchable database without forcing organizers to manually tag everything. You can use a regex-based extractor or a simple API call to a text-analysis tool. Better tagging directly leads to better search results and recommendations.

### PR DESCRIPTION
**Pull request by atul-upadhyay-7**
## What this fixes

When organizers create events on Uni-Event, they currently have to manually categorize events but have no way to add descriptive tags that improve search and discovery. The app now analyzes the event description as the organizer types and suggests relevant tags based on keyword matching.

## Root cause

The Create Event screen collected a title, description, and single category from the organizer, but had no mechanism to derive additional tags from the description content. Tags improve event discoverability in the feed and search results.

## What changed

- **app/src/lib/tagExtractor.js** — New service with a vocabulary of 24 tags mapped to domain-relevant keywords. Exports `extractTags(description)` which returns up to 5 matching tags sorted by keyword match count.
- **app/src/screens/CreateEvent.js** — Added `suggestedTags` and `selectedTags` state. A `useEffect` re-runs extraction whenever description changes. Suggested tags appear as tappable chips below the description field using the existing chip styling. Selected tags are saved to the Firestore `tags` field on the event document.
- **app/src/lib/__tests__/tagExtractor.test.js** — 12 unit tests covering keyword matching, case insensitivity, empty/null inputs, multiple matching categories, and the 5-tag limit.

## How to verify

1. Open the Create Event screen
2. Type a description like "Annual cultural fest with music, dance performances, and art exhibitions. Celebrate tradition."
3. Observe "Cultural Fest" appearing as a suggested tag chip below the description
4. Tap the chip to accept the tag — it highlights to show selection
5. Add more description text mentioning other topics and verify new suggestions appear
6. Create the event and check the Firestore document contains a `tags` array field

## Edge cases considered

- Empty/null description returns no suggestions
- Description with no matching keywords returns no suggestions
- Case-insensitive matching (e.g., "WORKSHOP" matches "Workshop")
- Response limited to 5 tags to avoid overwhelming the organizer
- Edit mode restores previously selected tags from the existing event document
- Multi-word keyword matching (e.g., "full stack" in "Web Dev")

Closes #72

## Summary by CodeRabbit

* **New Features**
  * Automatic tag suggestions when creating events, with selectable tag chips and saved event tags.
  * Improved daily digest delivery: respects opt-in, paginates sending, and returns richer delivery summary (including processed counts and no-events message).

* **Tests**
  * Added comprehensive tests for tag extraction covering multiple categories, edge cases, case-insensitivity, and result limits.

[Review Change Stack](https://app.coderabbit.ai/change-stack/roshankumar0036singh/Uni-Event/pull/287?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by atul-upadhyay-7** [comment]
@roshankumar0036singh could you review this

File: app/src/screens/CreateEvent.js [hunk 1]
```diff
@@ -155,10 +158,22 @@ export default function CreateEvent({ navigation, route }) {
...
+        setSelectedTags(prev =>
+            prev.includes(tag) ? prev.filter(t => t !== tag) : [...prev, tag],
+        );
+    };
+
     useEffect(() => {
         if (isEditMode) {
             setTitle(event.title);
             setDescription(event.description);
+            setSelectedTags(event.tags || []);
```
[the remark this answers is not in the document]
Comment (roshankumar0036singh) [reply app/src/screens/CreateEvent.js:176]: @atul-upadhyay-7 check this suggestion

**Review by roshankumar0036singh — APPROVED** [review]
